README.md

```diff
@@ -32,30 +28,52 @@ hygiene hidden `impl Drop` can be used by any API that would normally use a clos
 argument, such as crossbeam's scoped threads, to make the containing scope the safe,
 wrapped scope if they so desire the trade-off of macro versus closure indentation.
 
+It's important to note that lifetimes are only trusted to cary lifetime brands when they
+are in fact invariant. Variant lifetimes, such as `&'a T`, can be shrunk to fit the expected
+lifetime; `&'static T` can be used where `&'a T` is expected, for any `'a`.
+
 ## Informal proof of correctness
 
 ```rust
-let tag = unsafe { $crate::Id::new() };
-let $name = unsafe { $crate::Guard::new(tag) };
+let tag = unsafe { Id::new() };
+let $name = unsafe { Guard::new(tag) };
 let _guard = {
     #[allow(non_camel_case_types)]
-    struct make_guard<'id>(&'id $crate::Id<'id>);
-    impl<'id> ::core::ops::Drop for make_guard<'id> {
+    struct make_guard<'id> {
+        _id: Id<'id>,
+    }
+    impl<'id> Drop for make_guard<'id> {
         #[inline(always)]
         fn drop(&mut self) {}
     }
+    fn make_guard<'id>(_: &'id Id<'id>) -> make_guard<'id> {
+        make_guard { _id: *id }
+    }
     make_guard(&tag)
 };
 ```
 
-See also #1 for further discussion.
+### Disclaimer
+
+This relies on dead code (the empty drop) to impact borrow checking.
+Theoretically, a smarter CFG based borrow checker (i.e. NLL/polonius) *could*
+utilize the fact that this is dead code to remove this restriction, but this is
+*very* unlikely; the current (as of Rust 2021) NLL borrow checker requires dead
+code to be lifetime-correct, and this code isn't *dead* dead, as in, it *runs*,
+it just doesn't actually do anything other than impact lifetime solving. If you
+want to discuss the proof of correctness, the place to do so is issue #1.
```

Comment:
I would make `#1` an actual link to the issue.

Reply:
GitHub automatically linkifies #issue-number, but I suppose that yeah, it's nice to link it for e.g. crates.io.